# Downloads have been unreliable during the last few days; changing mirror.
diff --git a/modulesets-stable/gtk-osx-bootstrap.modules b/modulesets-stable/gtk-osx-bootstrap.modules
index 81b30ee..1f84cd5 100644
--- a/gtk-osx/modulesets-stable/gtk-osx-bootstrap.modules
+++ b/gtk-osx/modulesets-stable/gtk-osx-bootstrap.modules
@@ -9,7 +9,7 @@
   <repository type="tarball" name="sourceforge"
               href="http://downloads.sourceforge.net/sourceforge/"/>
   <repository type="tarball" name="jpeg"
-              href="http://www.ijg.org/files/"/>
+              href="https://fossies.org/"/>
   <repository type="tarball" name="libtiff"
 	      href="http://download.osgeo.org/"/>
   <repository type="tarball" name="icon-theme"
@@ -31,7 +31,7 @@
   </autotools>
 
   <autotools id="libjpeg" autogen-sh="configure">
-    <branch module="jpegsrc.v9d.tar.gz" version="9d"
+    <branch module="linux/misc/jpegsrc.v9d.tar.gz" version="9d"
             repo="jpeg" checkoutdir="jpeg-9d"
             hash="sha256:6c434a3be59f8f62425b2e3c077e785c9ce30ee5874ea1c270e843f273ba71ee">
     </branch>
# Add a meta package to pull in all dependencies without building gtk itself.
diff --git a/modulesets-stable/gtk-osx.modules b/modulesets-stable/gtk-osx.modules
index 76c32c4..7c5ac4c 100644
--- a/gtk-osx/modulesets-stable/gtk-osx.modules
+++ b/gtk-osx/modulesets-stable/gtk-osx.modules
@@ -314,6 +314,21 @@
     </after>
   </meson>
 
+  <metamodule id="meta-gtk3-dependencies">
+    <dependencies>
+      <dep package="glib"/>
+      <dep package="pango"/>
+      <dep package="atk"/>
+      <dep package="gdk-pixbuf"/>
+      <dep package="gobject-introspection"/>
+      <dep package="libepoxy"/>
+      <dep package="hicolor-icon-theme"/>
+    </dependencies>
+    <after>
+      <dep package="meta-gtk-osx-bootstrap"/>
+    </after>
+  </metamodule>
+
   <meson id="graphene" mesonargs="-Dtests=false">
     <branch repo="github-tarball" version="1.10.4"
             module="ebassi/graphene/releases/download/1.10.4/graphene-1.10.4.tar.xz"
